File: python/euler5.py
#! /usr/bin/env python

# projecteuler.net
#
# problem 5
#
# 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
#
# What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?

import pyprimes

limit = 20
p = []
y = []

for i in range(limit):
  if pyprimes.isprime(i):
    p.append(i)

for i in p:
	y.append(i)
	z = i * i
	if z < limit:
		while z < limit:
			z = z * i
			y.append(i)

print reduce(lambda a, b: a*b, y)

File: go/euler5.go
//
// projecteuler.net
//
// problem 5
//

package main

import (
	"fmt"
)

func main() {

	limit := 20
	p := [8]int{2, 3, 5, 7, 11, 13, 17, 19}
	y := []int{}
	z := 0

	for _, v := range p {
		y = append(y, v)
		z = v * v
		if z < limit {
			for z < limit {
				z = z * v
				y = append(y, v)
			}
		}
	}
	answer := y[0]
	for _, v := range y[1:] {
		answer *= v
	}
	fmt.Printf("%v\n", answer)
}
